projects/pswg/src/main/java/com/parzivail/pswg/entity/ThrowableExplosive.java

```diff
@@ -177,27 +176,23 @@ public boolean damage(DamageSource source, float amount)
 				this.delay = 2;
 				this.shouldExplode = true;
 			}
-
-			return true;
 		}
 		else if (source.isIn(SwgTags.DamageTypes.IGNITES_EXPLOSIVE))
 			if (!this.shouldExplode)
 				this.explode();
 
 		return super.damage(source, amount);
 	}
-
 	public void explode()
 	{
-		if (!this.getWorld().isClient)
+		if (!this.getWorld().isClient())
 		{
-			this.discard();
-			this.getWorld().createExplosion(this, (DamageSource)null, (ExplosionBehavior)null, this.getX(), this.getY(), this.getZ(), explosionPower, false, World.ExplosionSourceType.TNT, true);
+			getWorld().createExplosion(this, (DamageSource)null, (ExplosionBehavior)null, this.getX(), this.getY(), this.getZ(), explosionPower, false, World.ExplosionSourceType.TNT, false);
+			ServerWorld serverWorld = (ServerWorld)getWorld();
+			createParticles(getX(), getY(), getZ(), serverWorld);
 		}
-		if (this.getWorld() instanceof ServerWorld serverWorld)
-			createParticles(serverWorld, this.getX(), this.getY(), this.getZ());
+		discard();
```

Comment:
in general, make sure you're leaving blank lines between the end of a code block/braces and the next code